src/main/java/org/osc/manager/rest/server/api/SecurityApis.java

```diff
@@ -0,0 +1,173 @@
+import org.osc.manager.ism.api.IsmSecurityGroupApi;
+import org.osc.manager.ism.api.IsmSecurityGroupInterfaceApi;
+import org.osc.manager.ism.entities.SecurityGroupEntity;
+import org.osc.manager.ism.entities.SecurityGroupInterfaceEntity;
+import org.osc.manager.rest.server.SecurityManagerServerRestConstants;
+import org.osc.sdk.manager.element.ManagerSecurityGroupElement;
+import org.osc.sdk.manager.element.ManagerSecurityGroupInterfaceElement;
+import org.osc.sdk.manager.element.SecurityGroupInterfaceElement;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.transaction.control.TransactionControl;
+
+@Component(service = SecurityApis.class)
+@Path(SecurityManagerServerRestConstants.SERVER_API_PATH_PREFIX + "/securitygroups")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class SecurityApis {
+    private static Logger LOG = Logger.getLogger(SecurityApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+    private IsmSecurityGroupApi sgApi;
+    private IsmSecurityGroupInterfaceApi sgiApi;
+
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+        this.sgApi = new IsmSecurityGroupApi(null, txControl, em);
+        this.sgiApi = new IsmSecurityGroupInterfaceApi(null, null, txControl, em);
+    }
+
+    @POST
+    public String createSecurityGroup(SecurityGroupEntity entity) throws Exception {
+        LOG.info(String.format("Creating security group  with name %s", "" + entity.getName()));
+        return this.sgApi.createSecurityGroup(entity.getName(), null, null);
+    }
+
+    @Path("/{sgId}")
+    @PUT
+    public SecurityGroupEntity updateSecurityGroup(@PathParam("sgId") Long sgId, SecurityGroupEntity entity)
+            throws Exception {
+        LOG.info(String.format("Updating the security group for id %s ", "" + Long.toString(sgId)));
+        this.sgApi.updateSecurityGroup(Long.toString(sgId), entity.getName(), null);
+        return entity;
+    }
+
+    @Path("/{sgId}")
+    @DELETE
+    public void deleteSecurityGroup(@PathParam("sgId") Long sgId) throws Exception {
+        LOG.info(String.format("Deleting the security group for id %s ", "" + Long.toString(sgId)));
+        this.txControl.required(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                SecurityApis.this.sgApi.deleteSecurityGroup(Long.toString(sgId));
+                return null;
+            }
+        });
+    }
+
+    @GET
+    public List<String> getSecurityGroupIds() throws Exception {
+        LOG.info("Listing security group ids'");
+        List<? extends ManagerSecurityGroupElement> securityGroups = this.sgApi.getSecurityGroupList();
+        List<String> sgList = new ArrayList<String>();
+        if (securityGroups.isEmpty()) {
```

Comment:
invert the if to reduce the returns.
```
if (!sgs.isEmpty()) {  
sgList = securityGroups.stream().map(ManagerSecurityGroupElement::getSGId).collect(Collectors.toList()); 
} 
return sgList;
```

Reply:
Not relevant now, as that piece of code is removed as the list returns only one element.

Reply:
I missed, have incorporated this...